src/convertToObject.js

```diff
@@ -6,7 +6,38 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  // make an array with strings of sourceString
+  const properties = [];
+
+  let temp = sourceString.trim() + ';';
+
+  while (temp.includes(';')) {
+    properties.push(temp.slice(0, temp.indexOf(';') + 1));
+    temp = temp.slice(temp.indexOf(';') + 1).trim();
+  }
```

Comment:
[CODE PRACTICE] - Avoid using loops in this task, use iteration methods instead. Consider using `split` and `filter` to create an array of declarations.